.gitignore

```diff
@@ -0,0 +1,88 @@
+# ------------------------------------------------------------------------------
+# List of file patterns we do not want in the repo.
+# ------------------------------------------------------------------------------
+
+# Object and library files
+*.[oa]
+*.so
+*.dylib
+
+# Backup files
+*~
+
+# Totalview breakpoint files
+*.TVD.*breakpoints
+
+# Mac related
+*.dSYM
+.DS_Store
+
+# Generated files in the root MFEM directory
+/deps.mk
+
+# Generated files in config/
+config/_config.hpp
+config/config.mk
+
+# Optional user configuration files in config/
+config/user.cmake
+config/user.mk
+
+# Generated documentation files in doc/
+doc/CodeDocumentation.conf
+doc/CodeDocumentation.html
+doc/CodeDocumentation
+
+# Generated executables - examples and miniapps
+examples/**/ex[1-9]
+examples/**/ex[1-9]p
+examples/**/ex[1-9][0-9]
+examples/**/ex[1-9][0-9]p
+miniapps/electromagnetics/joule
+miniapps/electromagnetics/tesla
+miniapps/electromagnetics/volta
+miniapps/meshing/klein-bottle
+miniapps/meshing/mesh-explorer
+miniapps/meshing/mobius-strip
+miniapps/meshing/shaper
+miniapps/performance/ex[1-9]
+miniapps/performance/ex[1-9]p
+miniapps/tools/display-basis
```

Comment:
I can live with this list, but it is yet another place to update when we add new examples/miniapps...